packages/maskbook/src/plugins/ITO/UI/ClaimDialog.tsx

```diff
@@ -230,8 +230,8 @@ export function ClaimDialog(props: ClaimDialogProps) {
                             .dividedBy(ratio)
                             .multipliedBy(Math.pow(10, claimToken.decimals))
                         if (tAmount.isGreaterThan(maxSwapAmount)) return
-                        setTokenAmount(tAmount)
-                        setClaimAmount(tAmount.multipliedBy(ratio))
+                        setTokenAmount(tAmount.dp(0))
+                        setClaimAmount(tAmount.multipliedBy(ratio).dp(0))
```

Comment:
Only call `dp(0)` when set the state rather than on each BigNumber multiply or divide calculation, otherwise it would lose precision. If call `dp(0)` on every calculation when slide, you will find the slide undraggable bug solved https://github.com/DimensionDev/Maskbook/pull/2360, but the calculating result loses precision.